### ISSUE TITLE
Escape & in anchors

### ISSUE THREAD

**Issue by igalic** [Dalgona/Serum#135]
In "manually" tidying the generated HTML, i stumbled upon the following issues reported by tidy:

```
line 36 column 205 - Warning: unescaped & or unknown entity "&-voxpupuli"
line 37 column 1011 - Warning: unescaped & or unknown entity "&-s-it"
```

this tells me that we need to escape, or skip `&` when using it in anchors.

**Comment by Dalgona** [comment]
Thank you for your report! Where did you find those strings? In `id` attribute of HTML tags? For example:

```html
<h2 id="&-voxpupuli" ...>...</h2>
```

**Comment by igalic** [comment]
> Thank you for your report! Where did you find those strings? In `id` attribute of HTML tags? For example:

yup, although it looked more like:
 
```html
<h2 id="puppet-&-voxpupuli" ...>...</h2>
```

funnily enough, this happens whether my markdown says:

```markdown
## Puppet & Voxpupuli
```

or

```markdown
## Puppet &amp; Voxpupuli
```

**Comment by Dalgona** [comment]
Yes, that's becuase of how Serum generates element IDs based on text contents of `<h*>` tags. I'll find out how to properly normalize those ID values.

**Comment by Dalgona** [comment]
One thing to ask, what version of Serum are you using? I cannot reproduce this problem on my side. If you are using an old version, please update it and try again. (Better escaping of anchors will be implemented anyway, though.)

**Comment by igalic** [comment]
```elixir

  defp deps do
    [
      {:serum, git: "https://github.com/Dalgona/Serum.git", branch: "master"},
    ]
  end
```

**Comment by Dalgona** [comment]
I see, you _were_ on a bleeding edge! To make sure you're using the latest commit of Serum _v1_, Please try updating your deps to...

```elixir
defp deps do
  [
    {:serum, git: "https://github.com/Dalgona/Serum.git", branch: "v1/master"},
    # ...
  ]
end
```

Clean (or delete) your `deps` directory, fetch your dependency again using `mix deps.get`, and see if the problem still persists.

**Comment by igalic** [comment]
```
Your website is now ready!
Copy(or move) the contents of `site` directory
into your public webpages directory.
tidy -config tidy.config site/*.html || true
```

looking good.

**Comment by Dalgona** [comment]
Glad to hear that.

I'm leaving this thread open since there are some other quirks related to encoding HTML entities.
